src/USER-ADIOS/dump_atom_adios.cpp

```diff
@@ -0,0 +1,314 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+/* ----------------------------------------------------------------------
+   Contributing author: Norbert Podhorszki (ORNL)
+------------------------------------------------------------------------- */
+
+#include <string.h>
+#include "dump_atom_adios.h"
+#include "domain.h"
+#include "atom.h"
+#include "update.h"
+#include "group.h"
+#include "memory.h"
+#include "universe.h"
+#include "error.h"
+
+
+using namespace LAMMPS_NS;
+
+#define MAX_TEXT_HEADER_SIZE 4096
+#define DUMP_BUF_CHUNK_SIZE 16384
+#define DUMP_BUF_INCREMENT_SIZE 4096
+
+/* ---------------------------------------------------------------------- */
+
+DumpAtomADIOS::DumpAtomADIOS(LAMMPS *lmp, int narg, char **arg) :
+  DumpAtom(lmp, narg, arg)
+{
+    ad = new adios2::ADIOS("adios2_config.xml", world, adios2::DebugON);
+    groupSize = 0;
+}
+
+/* ---------------------------------------------------------------------- */
+
+DumpAtomADIOS::~DumpAtomADIOS()
+{
+    if (fh)
+    {
+        fh.Close();
+    }
+}
+
+/* ---------------------------------------------------------------------- */
+
+void DumpAtomADIOS::openfile()
+{
+    if (multifile) {
+        // if one file per timestep, replace '*' with current timestep
+        char *filestar = strdup(filename);
+        char *filecurrent = new char[strlen(filestar) + 16];
+        char *ptr = strchr(filestar,'*');
+        *ptr = '\0';
+        if (padflag == 0)
+            sprintf(filecurrent,"%s" BIGINT_FORMAT "%s",
+                    filestar,update->ntimestep,ptr+1);
+        else {
+            char bif[8],pad[16];
+            strcpy(bif,BIGINT_FORMAT);
+            sprintf(pad,"%%s%%0%d%s%%s",padflag,&bif[1]);
+            sprintf(filecurrent,pad,filestar,update->ntimestep,ptr+1);
+        }
+        fh = io.Open(filecurrent, adios2::Mode::Write, world);
+        if (!fh) {
+            char str[128];
+            sprintf(str,"Cannot open dump file %s",filecurrent);
+            error->one(FLERR,str);
+        }
+        free(filestar);
+        delete [] filecurrent;
+    }
```

Comment:
please use `} else {` and `if () {` and do not put the curly brace on the next line for consistency in programming style with the core of LAMMPS.

Reply:
Is there a specific formatting style that I can use in Eclipse to format LAMMPS source? Currently I format my codes with clang 3.8 with specifications for the ADIOS2 project (e.g. curly brace in new lines)